pkg/convert/tf.go

```diff
@@ -1299,10 +1314,12 @@ func rewriteTraversal(
 					newTraversal = append(newTraversal, hcl.TraverseRoot{Name: newName})
 					newTraversal = append(newTraversal, rewriteRelativeTraversal(scopes, "", traversal[1:])...)
 				} else {
-					// We don't know what this is, so lets assume it's an unknown resource (we shouldn't ever have unknown locals)
-					newName = scopes.getOrAddPulumiName(path, "", camelCaseName(root.Name))
```

Comment:
So I think the concern of _not_ using getOrAddPulumiName is that if the picked name here conflicts with some other bound or unbound name that's already in use we'll get a conflict in the PCL.
Feels like we could probably write a test case to cover that, and then fix this up.